src/main/java/com/javadiscord/jdi/internal/api/DiscordRequest.java

```diff
@@ -0,0 +1,98 @@
+package com.javadiscord.jdi.internal.api;
+
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublisher;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpRequest.Builder;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.javadiscord.jdi.internal.utils.UrlUtils;
+
+public class DiscordRequest {
+    public enum Method {
+        GET, POST, DELETE, PUT;
+    }
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private Method method;
+    private String url = "https://discord.com/api";
+    private Map<String, Object> headers;
+    private BodyPublisher body;
+    private DiscordResponse response;
+
+	public DiscordRequest(Method method, String path) {
```

Comment:
We have a lot of constructors here. Let's switch this and use the Builder Pattern.